From 3fd845cf506a48e779087b8d2453e2a156e208c5 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Mon, 11 Dec 2023 09:20:07 -0800
Subject: [PATCH 1/6] pytorch_rocm preconfig, build and install scripts

- clean previous build, build wheel and install wheel scripts
"-Wno-error=maybe-uninitialized" is needed during
build time for gcc14/fedora 40.

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 build_rocm.sh     | 23 +++++++++++++++++++++++
 install_rocm.sh   | 23 +++++++++++++++++++++++
 preconfig_rocm.sh | 21 +++++++++++++++++++++
 3 files changed, 67 insertions(+)
 create mode 100755 build_rocm.sh
 create mode 100755 install_rocm.sh
 create mode 100755 preconfig_rocm.sh

diff --git a/build_rocm.sh b/build_rocm.sh
new file mode 100755
index 0000000000..8427d3549a
--- /dev/null
+++ b/build_rocm.sh
@@ -0,0 +1,23 @@
+if [ -z "$1" ]; then
+    install_dir_prefix_rocm=/opt/rocm
+    echo "No rocm_root_directory parameter specified, using default: ${install_dir_prefix_rocm}"
+else
+    install_dir_prefix_rocm=${1}
+    echo "rocm_root_directory parameter: ${install_dir_prefix_rocm}"
+fi
+if [ -z "$2" ]; then
+    rocm_version_str=60101
+    echo ""
+    echo "Error, no rocm version parameter specified. Usage example for amd rocm version 6.1.1:"
+    echo "./build_rocm.sh /opt/rocm_sdk_600 ${rocm_version_str}"
+    echo ""
+    exit 1
+else
+    rocm_version_str=${2}
+    echo "Using rocm version parameter: ${rocm_version_str}"
+fi
+unset LDFLAGS
+unset CFLAGS
+unset CPPFLAGS
+unset PKG_CONFIG_PATH
+USE_FLASH_ATTENTION=OFF ROCM_PATH=${install_dir_prefix_rocm} ROCM_SOURCE_DIR=${install_dir_prefix_rocm} CMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS -Wno-error=maybe-uninitialized" CMAKE_PREFIX_PATH="${install_dir_prefix_rocm};${install_dir_prefix_rocm}/lib64/cmake;${install_dir_prefix_rocm}/lib/cmake;${install_dir_prefix_rocm}/lib64;${install_dir_prefix_rocm}/lib" ROCM_VERSION=${rocm_version_str} HIP_ROOT_DIR=${install_dir_prefix_rocm} USE_ROCM=1 python setup.py bdist_wheel
diff --git a/install_rocm.sh b/install_rocm.sh
new file mode 100755
index 0000000000..38ed0fc21d
--- /dev/null
+++ b/install_rocm.sh
@@ -0,0 +1,23 @@
+unset backup_dir_python_wheel
+if [ ! -z "$1" ]; then
+    backup_dir_python_wheel=${1}
+    echo "copying pytorch python whl file to directory: ${backup_dir_python_wheel}"
+fi
+# if there are multiple wheel files, find the newest one and install it
+unset -v latest_wheel_file
+for cur_file in dist/*.whl; do
+    [[ $cur_file -nt "$latest_wheel_file" ]] && latest_wheel_file=$cur_file
+done
+if [ ! -z "$latest_wheel_file" ]; then
+    echo "installing $latest_wheel_file"
+    pip install "$latest_wheel_file"
+    if [ ! -z "$backup_dir_python_wheel" ]; then
+        if [ ! -d "$backup_dir_python_wheel" ]; then
+            mkdir -p "$backup_dir_python_wheel"
+        fi
+        cp -f "$latest_wheel_file" "$backup_dir_python_wheel"
+    fi
+else
+    echo "Could not find pytorch whl file to install"
+    exit 1
+fi
diff --git a/preconfig_rocm.sh b/preconfig_rocm.sh
new file mode 100755
index 0000000000..92aacb9f0c
--- /dev/null
+++ b/preconfig_rocm.sh
@@ -0,0 +1,21 @@
+if [ -z "$1" ]; then
+    install_dir_prefix_rocm=/opt/rocm
+    echo "No rocm_root_directory_specified, using default: ${install_dir_prefix_rocm}"
+else
+    install_dir_prefix_rocm=${1}
+    echo "using rocm_root_directory specified: ${install_dir_prefix_rocm}"
+fi
+unset LDFLAGS
+unset CFLAGS
+unset CPPFLAGS
+unset PKG_CONFIG_PATH
+if [ -e ./preconfig_rocm.sh ]; then
+    if [ -d ./build ]; then
+        #rm -rf torch (this is needed to really get all files regenerated for hip)
+        #git status | xargs -- rm -rf
+        #git reset --hard
+        #git submodule update --init --recursive
+        python setup.py clean
+    fi
+fi
+USE_FLASH_ATTENTION=OFF CMAKE_PREFIX_PATH="${install_dir_prefix_rocm};${install_dir_prefix_rocm}/lib64 python" tools/amd_build/build_amd.py
-- 
2.45.2

